flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1379,15 +1399,17 @@ struct XEmboxOpConversion : public EmboxCommonConversion<fir::cg::XEmboxOp> {
       if (hasSlice)
         sliceOff += 3;
     }
-    auto hasSubcomp = !xbox.subcomponent().empty();
     if (hasSlice || hasSubcomp) {
       llvm::SmallVector<mlir::Value> args = {base, ptrOffset};
       args.append(gepArgs.rbegin(), gepArgs.rend());
       if (hasSubcomp) {
         // For each field in the path add the offset to base via the args list.
-        // In the most general case, some offsets must be computed since they
-        // are not be known until runtime.
-        TODO(loc, "intra-entity slice in fir.embox codegen");
+        // TBD: In the most general case, some offsets must be computed since
+        // they are not be known until runtime.
```

Comment:
I added a TBD to the above comment. I'm not sure what was meant by some offsets are not known until runtime. Maybe this comment should be removed?

Reply:
This plan is news to me.  Is there a design document anywhere for it?  How will the runtime find these components?  (At present, it implements them as automatic objects.)